lib/application/pages/home/collections/update/update_providers.dart

```diff
@@ -0,0 +1,18 @@
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:memo/application/view-models/home/update_collection_vm.dart';
+
+/// Overridable collection id used in the scope of a collection update.
+final updateCollectionId = Provider<String?>((_) => throw UnimplementedError(), name: 'updateCollectionId');
+
+/// Syntax sugar for listening to [UpdateCollectionState] state updates.
+UpdateCollectionState useUpdateCollectionState(WidgetRef ref) {
+  final collectionId = ref.watch(updateCollectionId);
+  return ref.watch(updateCollectionVM(collectionId));
+}
+
+/// Syntax sugar for `ref.watch` the current [UpdateCollectionVM] provider.
+UpdateCollectionVM useUpdateCollectionVM(WidgetRef ref) {
```

Comment:
```suggestion
UpdateCollectionState watchUpdateCollectionState(WidgetRef ref) {
  final collectionId = ref.watch(updateCollectionId);
  return ref.watch(updateCollectionVM(collectionId));
}

/// Syntax sugar for `ref.watch` the current [UpdateCollectionVM] provider.
UpdateCollectionVM watchUpdateCollectionVM(WidgetRef ref) {
```

Can't we also make this a non-family provider? wdyt?

Reply:
I made this changes in a [future PR](https://github.com/olmps/memo/pull/237/files#diff-35ff452a74be7334307fbfa02595615c1d02eec30fc17b6158156376324ce851R1-R10) after the discussion we had in Discord. Please ignore this family providers and functions for now

Reply:
I removed this auxiliary functions and the `family` providers in a [future PR](https://github.com/olmps/memo/pull/237/files#diff-35ff452a74be7334307fbfa02595615c1d02eec30fc17b6158156376324ce851R1-R10). Please ignore them by now